docs: document maintenance window check behavior (#281)
checkly/docs: communicate/maintenance-windows/overview.mdx

---
title: 'Maintenance Windows'
description: 'Schedule planned maintenance periods to prevent false alerts and control check behavior during downtime.'
sidebarTitle: 'Maintenance Windows'
---

<Tip>
**Monitoring as Code**: Learn more about the [Maintenance Window Construct](/constructs/maintenance-window).
</Tip>

Maintenance windows allow you to schedule planned downtime for your services, preventing false alerts and preserving accurate availability metrics during system updates, deployments, and infrastructure maintenance activities.

During a maintenance window you can independently control two behaviors:
- **Pause check execution** — scheduled check runs are skipped entirely.
- **Silence alerts** — checks keep running and collecting data, but alert channel notifications (email, Slack, etc.) are suppressed.

> You can enable either behavior, both, or neither depending on what you need.

Reviewer: ```suggestion
You can enable a combination of these settings depending on what you need.
```

> If you schedule a window on the 31st and add a **monthly repeat cycle**, your schedule will also activate on
> the 30th of the relevant months and the 28th or 29th of February depending on the leap year.

### Check behavior

> Each maintenance window lets you configure how checks behave during the maintenance period. Toggle the behaviors you need:

Reviewer: Looks like we're repeating what's already said above. Shouldn't we only keep a single copy?

Author: Yeah that's true, removed

#### Pause check execution

When enabled, scheduled check runs are skipped for the duration of the window. You can scope which checks are paused:

- **All checks** — every check in the account is paused.
- **Checks matching tags** — only checks (and check groups) with matching tags are paused.

#### Silence alerts

When enabled, alert channel notifications are suppressed while checks continue to run and collect data. This is useful when you want to keep monitoring but avoid noisy alerts during expected degradation. You can scope which checks are silenced:

- **All checks** — alerts are silenced for every check in the account.
- **Checks matching tags** — only checks (and check groups) with matching tags have their alerts silenced.

<Warning>

> If a check starts failing during the maintenance window and continues failing after the window ends, Checkly will **not** retroactively send an alert once the window is over — the alert is suppressed. Enable **Firewatch** on your home dashboard to surface checks that started failing during maintenance.

Reviewer: Link to Firewatch and how to enable it?

Author: I can't find any docs on Firewatch, but I can add a description on where to find it

</Warning>

<Tip>
You can use different tag sets for pausing and silencing. For example, pause your `api` checks while silencing alerts for `frontend` checks that might be affected by the same deployment.
</Tip>

### Adding checks via tags

You scope which checks are affected by adding tags to the pause or silence configuration. Any checks or check groups with matching tags will be included.

This way, any checks or check groups created *after* creating your maintenance window can still be made part of the window; no
need to update the maintenance window configuration. Just set the correct tags.

## Monitoring maintenance status

> During an active maintenance window, the home dashboard shows:

Reviewer: If time I'd add a screenshot, but that could also be a follow up

Author: Once the other bigger changes are out, I will make another PR for the docs and include screenshots then

Reviewer: Yeah totally not crucial. It's also more maintenance (haha) burden to keep screenshots up to date, free of PII and still relevant, etc.
